rust/lance-namespace/src/rest.rs

```diff
@@ -0,0 +1,547 @@
+        // Note: This assumes index_name is provided somehow - in the Python version it's a separate parameter
+        // For simplicity, we'll assume it's in the request somewhere or use a default
+        let index_name = "default"; // This should be properly handled in a real implementation
+        
+        let path = format!("tables/{}/indices/{}/stats?delimiter={}", id_str, index_name, urlencoding::encode(&self.config.delimiter()));
+        let body = serde_json::to_vec(&request)?;
+        
+        self.make_request(reqwest::Method::POST, &path, Some(body)).await
+    }
+    
+    async fn describe_transaction(&self, request: DescribeTransactionRequest) -> Result<DescribeTransactionResponse> {
+        let id_str = self.object_id_str(request.id.as_deref().unwrap_or_default());
+        
+        let path = format!("transactions/{}?delimiter={}", id_str, urlencoding::encode(&self.config.delimiter()));
+        let body = serde_json::to_vec(&request)?;
+        
+        self.make_request(reqwest::Method::POST, &path, Some(body)).await
+    }
+    
+    async fn alter_transaction(&self, request: AlterTransactionRequest) -> Result<AlterTransactionResponse> {
+        let id_str = self.object_id_str(request.id.as_deref().unwrap_or_default());
+        
+        let path = format!("transactions/{}?delimiter={}", id_str, urlencoding::encode(&self.config.delimiter()));
+        let body = serde_json::to_vec(&request)?;
+        
+        self.make_request(reqwest::Method::PATCH, &path, Some(body)).await
+    }
+}
+
+pub struct RestNamespaceConfig {
+    delimiter: String,
+    headers: HashMap<String, String>,
+    uri: Option<String>,
+}
+
+impl RestNamespaceConfig {
+    const DELIMITER: &'static str = "delimiter";
+    const HEADER_PREFIX: &'static str = "header.";
+    const URI: &'static str = "uri";
+    
+    pub fn new(properties: HashMap<String, String>) -> Result<Self> {
+        let delimiter = properties.get(Self::DELIMITER).cloned().unwrap_or_else(|| ".".to_string());
+        let uri = properties.get(Self::URI).cloned();
+        let headers = Self::extract_headers(&properties);
+        
+        Ok(RestNamespaceConfig {
+            delimiter,
+            headers,
+            uri,
+        })
+    }
+    
+    fn extract_headers(properties: &HashMap<String, String>) -> HashMap<String, String> {
+        let mut headers = HashMap::new();
+        for (key, value) in properties {
+            if key.starts_with(Self::HEADER_PREFIX) {
+                let header_name = key.strip_prefix(Self::HEADER_PREFIX).unwrap();
+                headers.insert(header_name.to_string(), value.clone());
+            }
+        }
+        headers
+    }
+    
+    pub fn delimiter(&self) -> &str {
+        &self.delimiter
+    }
+    
+    pub fn additional_headers(&self) -> &HashMap<String, String> {
+        &self.headers
+    }
+    
+    pub fn uri(&self) -> Option<String> {
+        self.uri.clone()
+    }
+}
+
+#[cfg(test)]
+mod tests {
```

Comment:
Consult tests in https://github.com/lancedb/lancedb/blob/main/rust/lancedb/src/remote/db.rs for how we can run tests with mock response.